hi-ml-histopathology/src/histopathology/utils/analysis_plot_utils.py

```diff
@@ -267,3 +267,34 @@ def add_training_curves_legend(fig: Figure, include_best_epoch: bool = False) ->
                                      color='k', label="Best epoch (val.)"),)
     fig.legend(handles=legend_handles, **legend_kwargs, loc='lower center',
                bbox_to_anchor=(0.5, -0.1), ncol=len(legend_handles))
+
+
+def plot_confusion_matrices(crossval_dfs: Dict[int, pd.DataFrame], class_names: List[str]) -> Figure:
+    """
+    Plot normalized confusion matrices from HyperDrive child runs.
+    :param crossval_dfs: Dictionary of dataframes with cross-validation indices as keys,
+        as returned by :py:func:`collect_crossval_outputs()`.
+    :param class_names: Names of classes.
+    :return: The created `Figure` object.
+    """
+    import seaborn as sns
+    fig, axs = plt.subplots(1, len(crossval_dfs), figsize=(len(crossval_dfs) * 6, 5))
+    for k, tiles_df in crossval_dfs.items():
+        slides_groupby = tiles_df.groupby(ResultsKey.SLIDE_ID)
+        tile_labels_true = slides_groupby[ResultsKey.TRUE_LABEL]
+        # True slide label is guaranteed unique
+        assert all(len(unique_slide_label) == 1 for unique_slide_label in tile_labels_true.unique())
+        labels_true = tile_labels_true.first()
+
+        tile_labels_pred = slides_groupby[ResultsKey.PRED_LABEL]
+        labels_pred = tile_labels_pred.first()
+
+        cf_matrix = confusion_matrix(y_true=labels_true, y_pred=labels_pred)
+        cf_matrix_n = cf_matrix / cf_matrix.sum(axis=1, keepdims=True)
+        sns.heatmap(cf_matrix_n, annot=True, cmap='Blues', fmt=".2%", ax=axs[k])
+        axs[k].set_xlabel('Predicted')
+        axs[k].set_ylabel('True')
+        axs[k].xaxis.set_ticklabels(class_names)
```

Comment:
[Minor] You could pass these straight to `sns.heatmap()` as `xticklabels` and `yticklabels` arguments ([docs](https://seaborn.pydata.org/generated/seaborn.heatmap.html)).

Reply:
Thanks, done